src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static int GROWTH_FACTOR;
```

Comment:
The `GROWTH_FACTOR` is declared as a static variable, which means it will be shared across all instances of `MateStack`. This can lead to unexpected behavior if different instances are initialized with different growth factors. Consider making it a non-static instance variable.